src/ophyd_async/epics/pvi/pvi.py

```diff
@@ -158,28 +158,34 @@ def _sim_common_blocks(device: Device, stripped_type: Optional[Type] = None):
             (origin := get_origin(sub_device_t)) and issubclass(origin, Signal)
         ) or (issubclass(sub_device_t, Signal))
 
-        if is_device_vector and is_signal:
-            signal_type = args[0] if (args := get_args(sub_device_t)) else None
-            sub_device_1 = sub_device_t(SimSignalBackend(signal_type, sub_name))
-            sub_device_2 = sub_device_t(SimSignalBackend(signal_type, sub_name))
-            sub_device = DeviceVector(
-                {
-                    1: sub_device_1,
-                    2: sub_device_2,
-                }
-            )
-        elif is_device_vector and not is_signal:
-            sub_device = DeviceVector(
-                {
-                    1: sub_device_t(name=f"{device.name}-{sub_name}-1"),
-                    2: sub_device_t(name=f"{device.name}-{sub_name}-2"),
-                }
-            )
+        # TODO: worth coming back to all this code once 3.9 is gone and we can use
```

Comment:
Make an issue for removing 3.9 support, refactoring this code and link in the TODO

Reply:
There is already an issue. I'll link it

Reply:
https://github.com/bluesky/ophyd-async/issues/180

Reply:
Can you link it in the Python comment, so the TODO can be followed up easier?